### DISCUSSION TITLE
how can I pass arguments to pip from poetry?

### DISCUSSION THREAD

**Discussion by jtmoon79** [python-poetry/poetry#3909] [Q&A]
How can I pass arguments to child process invocations of `pip`? Is there a free form field in `pyproject.toml` that allows this?

For example, how would I pass `--disable-pip-version-check` to all `pip` calls?
Another example, how would I pass `--trusted-host hostname` ?

[Also asked on StackOverflow](https://stackoverflow.com/questions/67028043/arguments-to-pip-should-be-passed-from-tox-or-poetry-or-shell).

**Comment by abn** [comment]
This should no longer be required as poetry does not download wheels via pip, in almost all cases; poetry calls pip similar to this:

```
pip install --disable-pip-version-check --prefix /path/to/env/site --no-deps file://../dep.whl
```

This is because poetry manages the download itself, into its own cache.

Exceptions for this right now are:
1. Source installs. These are similar to above but will use `--use-pep517`.
2. PEP 517 build requirements. These will be bootstrapped by pip.

These exceptions will eventually be handled outside of pip as well. 

Ref: https://github.com/python-poetry/poetry/blob/bf18a8f700c9b717e9ab36571a2ef3010a09132a/poetry/utils/pip.py#L26

**Reply by jtmoon79** [reply]
Thanks @abn 

Perhaps you have advice for my particular problem.

In my case, I'm running into a problem during CI runs, `tox` calls `poetry update` which fails with (selective output)
```
Stack trace:
...
  3  .tox/diff/lib/python/site-packages/poetry/installation/executor.py in run_pip
     self._env.run_pip(*args, **kwargs)
...
WARNING: The repository located at pypi-internal.local is not a trusted or secure host and is being ignored. If this repository is available via HTTPS we recommend you use HTTPS instead, otherwise you may silence this warning and allow it anyway with '--trusted-host pypi-internal.local'.
ERROR: Could not find a version that satisfies the requirement some-package~=1.0.1
```
How do I instruct `poetry` to instruct `pip` to add trust for host `pypi-internal.local` ?

**Reply by jtmoon79** [reply]
Oh wait, maybe I need to update `poetry` to `1.1.5`

https://github.com/python-poetry/poetry/pull/3622/files#diff-e47bc2b1dc22a4c9a701ba19a730976bf2011e80077df4edd27dec2fb25eeafcR144-R145

**Reply by abn** [reply]
That one is for the exporter; so when you run `poetry export`. But that said, having a more recent version can help. Since the new installer should not download wheels using pip when you run `poetry install`.

**Reply by abn** [reply]
In the interim however, you could try doing something like this to influence pip.

```console
$ cat ~/.pip/pip.conf 
[global]
trusted-host = pypi-internal.local
```
